tools/runners/run-test-suite.sh

```diff
@@ -168,9 +172,11 @@ do
         status_code=$?
     fi
 
+    testfile=$(basename $(echo $cmd_line | sed s/" "/"\n"/g | tail -1))
     if [ $status_code -ne $error_code ]
     then
-        echo "[$tested/$TOTAL] $cmd_line: FAIL ($status_code)"
+        printf "[%4d/%4d] " "$tested" "$TOTAL"
+        echo -e "${TERM_RED}FAIL ($status_code): $testfile $TERM_EMPTY"
```

Comment:
There is no need for mixing `printf`s and `echo`s. If `printf` is the tool of choice then let's print that log line with it. Note: `printf` can deal with escape sequences with the help of `%b%`. E.g., `printf "xxx%byyy%bzzz\n" "\e[31m" "\e[0m"` (the y's in the middle should appear red).